### ISSUE TITLE
add to kubectl explain config APIs

### ISSUE THREAD

**Issue by 111andre111** [kubernetes/kubernetes#127000]
### What would you like to be added?

At the moment `kubectl explain` only allows to explain all resource that are returned by `kubectl api-resources`.
I'd like seeing to add the configuration APIs as well to explain.

### Why is this needed?

When creating kubelet configuration files or other external resources like kubeadm config files or pod-security files and it would be easier to spin up `kubectl explain` rather than going to kubernetes docs and read the provided documentation there: https://kubernetes.io/docs/reference/config-api/

**Comment by neolit123** [comment]
> ### What would you like to be added?
> At the moment `kubectl explain` only allows to explain all resource that are returned by `kubectl api-resources`. I'd like seeing to add the configuration APIs as well to explain.
> 
> ### Why is this needed?
> When creating kubelet configuration files or other external resources like kubeadm config files or pod-security files and it would be easier to spin up `kubectl explain` rather than going to kubernetes docs and read the provided documentation there: https://kubernetes.io/docs/reference/config-api/

i don't think this is feasible with stock kubectl explain, but maybe a plugin can do it.
this technically makes the feature request external to kubernetes/kubernetes.

/sig cli

**Comment by 111andre111** [comment]
Ok, then would that mean I should transfer this Isabell over to kubernetes?

**Comment by neolit123** [comment]
SIG CLI (the group which i assigned) would provide feedback.

**Comment by 111andre111** [comment]
Ok, so if I understand correctly I don't need to do anything from my side?

**Comment by Jefftree** [comment]
Not sure how feasible this is. `kubectl explain` is used for api resources, only ones that CRUD operations can be performed on via the k8s rest API. I don't disagree that it would be useful to have this somewhere but explain may not be the best spot for it.

**Comment by 111andre111** [comment]
If that's not the case, I am open for suggestions. It was always difficult to create config resource files given you need to manually obtain the information from docs in comparison you would develop a yaml file of an api resources.

**Comment by ah8ad3** [comment]
Thanks for the idea @111andre111

I believe this needs more discussions, if you like you can ask in #sig-cli slack channel or maybe discuss this at the regular sig-cli meeting.

**Comment by ardaguclu** [comment]
> Not sure how feasible this is. `kubectl explain` is used for api resources, only ones that CRUD operations can be performed on via the k8s rest API. I don't disagree that it would be useful to have this somewhere but explain may not be the best spot for it.

I agree with to @Jefftree. The recommended path for such cases is to implement a plugin.
